specs/AnimatedVisualPlayer Spec.md

```diff
@@ -0,0 +1,145 @@
+|Always| `AnimatedVisualPlayer` will pre-create animations even before `PlayAsync()` is called,
+and not destroy any when player is idle. |
+|None | `AnimatedVisualPlayer` will not create animations until `PlayAsync()` is called,
+and will destroy them when it completes. |
+
+If you have an animation that will start on click, you may want to set `CacheMode` to `Always`
+when the mouse enters this control, 
+and back to `None` when mouse leaves to ensure that the animation starts immediately on click,
+but don't use up resources otherwise.
+
+Setting the `CacheMode` to `Always` when the mouse enters the control will ensure that
+the animation objects are preloaded so that the animation may start immediately when it needs to play. 
+
+If your player is does not have any animations to play and `CacheMode` is set to `Always`, the player will instantiate all animations immediately and play what was instantiated. 
+
+```c#
+private void Player_PointerEntered(object sender, PointerRoutedEventArgs e)
+{
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.Always;
+}
+async private void Player_PointerExited(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.CacheMode = AnimationsCacheMode.None;
+}
+
+async private void Player_OnClick(object sender, PointerRoutedEventArgs e)
+{              
+    myanimatedvisualplayer.PlayAsync(0.0, 1.0);
+}
+
+```
+
+If `PlayAsync` is called and the animations have not been instantiated due to `CacheMode` being set to `None`,
+`PlayAsync` will instantiate the animations so it can play the visual. 
+
+# API Details
+
+```c#
+namespace Microsoft.UI.Xaml.Controls
+{
+  unsealed runtimeclass AnimatedVisualPlayer
+  {
+      // Existing ...
+
+    //Sets the caching behavior of the AnimatedVisualPlayer
+    AnimationsCacheMode CacheMode;
+  }
+
+  public enum AnimationsCacheMode
+  {
+      Always = 0,
+      None
+  }
+}
+```
+
+## IAnimatedVisual2 Interface
+
+An extension to
+[IAnimatedVisual](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisual)
+that adds the ability to control when animation objects are created and destroyed.
+
+| Method | Meaning |
+|-------------|---------|
+| InstantiateAnimations | Instantiates the animation visuals for the provided progress value of the animation. |
+| DestroyAnimations | Destroys all animation visuals for a given animation.|
+
+```c# 
+interface IAnimatedVisual2 requires IAnimatedVisual
+{
+    void InstantiateAnimations(Double progressHint);
+    void DestroyAnimations();
+};
+```
+
+## IAnimatedVisualSource3 Interface
+
+An extension to
+[IAnimatedVisualSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.IAnimatedVisualSource)
+that products an `IAnimatedVisualSource3`.
+
+```c# 
+interface IAnimatedVisualSource3
+{
+    IAnimatedVisual2 TryCreateAnimatedVisual(
+        Windows.UI.Composition.Compositor compositor,
+        out Object diagnostics,
+        Boolean instantiateAnimations,);
+};
```

Comment:
Should note "We need a new method for this, rather than just using `IAnimatedVisualSource.TryCreateAnimatedVisual` because we need the ability to create an animation without instantiating all the animations. The old method returns an animated visual source that is fully-instantiated. We can't use `IAnimatedVisualSource.TryCreatAnimatedVisual()` + `IAnimatedVisual2.DestroyAnimations()` because that would instantiate the animations, only to destroy them immediately.

Reply:
Yes, we shouldn't use `IAnimatedVisualSource.TryCreatAnimatedVisual()` if animation implements `IAnimatedVisualSource3` interface, but if we are detecting that animation implements `IAnimatedVisualSource3` we will call `IAnimatedVisualSource3.TryCreateAnimatedVisual`  and this one will have a flag to disable animations creation.

Reply:
Maybe a general note (on each of the `TryCreateAnimatedVisual` pages) to be aware that there are multiple to choose from and briefly why.